portal/serializers.py

```diff
@@ -0,0 +1,89 @@
+from rest_framework import serializers
+
+from portal.models import Poll, PollOption, PollVote
+
+
+class UserPollSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Poll
+        fields = ("id", "source", "question", "expire_date", "admin_comment")
+
+    def create(self, validated_data):
+        # adds user to the Poll
+        instance = Poll(**validated_data, user=self.context["request"].user)
+        # ensuring user cannot create an admin comment upon creation
+        instance.admin_comment = ""
+        instance.save()
+        return instance
+
+    def update(self, instance, validated_data):
+        user = self.context["request"].user
+        # checks if user is allowed to update or not
+        if user != instance.user:
+            raise serializers.ValidationError(detail={"detail": "User not allowed to update."})
+        return super().update(instance, validated_data)
+
+
+class AdminPollSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Poll
+        fields = "__all__"
+
+
+class PollOptionSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = PollOption
+        fields = (
+            "id",
+            "poll",
+            "choice",
+        )
+
+    def update(self, instance, validated_data):
+        user = self.context["request"].user
+        # checks to see if user can update the option
+        if user != instance.poll.user:
+            raise serializers.ValidationError(detail={"detail": "User not allowed to update."})
+        return super().update(instance, validated_data)
+
+
+class RetrievePollSerializer(serializers.ModelSerializer):
+
+    options = PollOptionSerializer(source="polloption_set", many=True)
```

Comment:
does this need a `read_only=True` or something similar? I thought django tends to complain if you don't do that

Reply:
it did not throw an error for me after testing haha. should i put it just in case?